rest-api/test/unit_test/api_test/participant_summary_api_test.py

```diff
@@ -249,6 +252,129 @@ def testQuery_noSummaries(self):
+    self.assertEqual(
+      sorted([p['participantId'] for p in first_batch]),
+      sorted([p['resource']['participantId'] for p in response['entry']]),
+    )
+    self.assertEqual(sorted([p['resource']['lastModified'] for p in response['entry']]),
+                      [p['resource']['lastModified'] for p in response['entry']])
+    self.assertEqual(response['entry'][0]['resource']['lastModified'], t1.strftime(
+      '%Y''-''%m''-''%d''T''%X'))
+    self.assertEqual(response['entry'][1]['resource']['lastModified'], t1.strftime(
+      '%Y''-''%m''-''%d''T''%X'))
+    self.assertEqual(response['entry'][5]['resource']['lastModified'], t2.strftime(
+      '%Y''-''%m''-''%d''T''%X'))
+    self.assertEqual(response['entry'][7]['resource']['lastModified'], t3.strftime(
+      '%Y''-''%m''-''%d''T''%X'))
+
+    # Get the next chunk with the sync url
+    # Verify that this is, in fact, a sync URL - not a next
+    sync_url = response['link'][0]['url']
+    index = sync_url.find('ParticipantSummary')
+    self.assertEqual(response['link'][0]['relation'], 'sync')
+
+    # Verify that the next sync has results from t2 and t3 (within BUFFER).
+    response2 = self.send_get(sync_url[index:])
+    self.assertEqual(len(response2['entry']), 5)
+
+    # Create a second batch
+    second_batch = [setup_participant(t4) for _ in range(10)]
+    response3 = self.send_get(sync_url[index:])
+    # We have the same number of participants as summaries
+    self.assertEqual(len(response3['entry']), len(second_batch) + len(response2['entry']))
+
+    no_count_url = 'ParticipantSummary?lastModified=lt%s&_sync=true&awardee=PITT' % TIME_4
+    no_count_response = self.send_get(no_count_url)
+    total_count = len(no_count_response['entry'])
+    self.assertEquals(total_count, 20)
+    url = 'ParticipantSummary?lastModified=lt%s&_count=10&_sync=true&awardee=PITT' % TIME_4
+    response = self.send_get(url)
+    self.assertEqual(len(response['entry']), 10)
+    next_url = response['link'][0]['url']
+    next_10 = self.send_get(next_url[index:])
+    self.assertEqual(len(next_10['entry']), 10)
+
+    sort_by_lastmodified = 'ParticipantSummary?_sync=true&awardee=PITT&_sort=lastModified'
+    sort_lm_response = self.send_get(sort_by_lastmodified)
+    self.assertEquals(len(sort_lm_response['entry']), 20)
+    self.assertEquals(sort_lm_response['link'][0]['relation'], 'sync')
+    # ensure same participants are returned before 5 min. buffer
+    sync_url = sort_lm_response['link'][0]['url']
+    setup_participant(t5)
+    sync_again = self.send_get(sync_url[index:])
+    self.send_get(sort_by_lastmodified)
+    self.assertEquals(len(sync_again['entry']), 14)
+    one_min_modified = list()
+
+    for i in sync_again['entry']:
+      one_min_modified.append(datetime.datetime.strptime(i['resource']['lastModified'],
+                                                            '%Y''-''%m''-''%d''T''%X'))
+
+    # Everything should be within 60 seconds.
```

Comment:
You could alternatively assert the exact timestamps / IDs that come back.

(Creating fewer participants may make this simpler.)

Reply:
done, in a super clever way 😄

Reply:
Sorry I don't see that change?

Reply:
Line 403 asserts that every participant object is the same for the last 14 elements of `sort_lm_response` and `sync_again`.

Reply:
OK. Then arguably everything after line 413 is unnecessary?